### PR TITLE
Create contributing guide

### LINKED ISSUE #8: How to debug Quarkus JDT LS Extension?
Explain how to debug Quarkus JDT LS Extension

### PR DESCRIPTION
**Pull request by xorye**
Fixes [quarkus-ls #8](https://github.com/redhat-developer/quarkus-ls/issues/8). 

CONTRIBUTING.md is the first draft for the contributing guide. 
In the markdown file, I briefly explain how communication works, how to build the projects and how to debug.

Please let me know if anything could be explained better, or if anything should be added.

Signed-off-by: David Kwon <dakwon@redhat.com>

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+The image above represents communication between the three components. 
+It is important to note that the Quarkus Language Server cannot directly 
+communicate with the Quarkus jdt.ls extension and vice-versa. They must 
+communicate via vscode-quarkus.
+  
+Step 1. A Quarkus project is open is VSCode, and completion has been 
+invoked inside the `application.properties` file, which sends a
+`textDocument/completion` request to the Quarkus language server.  
+
+Step 2. Quarkus langauge server checks its cache if completion options
```
Comment (fbricon) [inline CONTRIBUTING.md]: language

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+for the current Quarkus project exists.
```
Comment (fbricon) [inline CONTRIBUTING.md]: (completion options) exist

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+invoked inside the `application.properties` file, which sends a
+`textDocument/completion` request to the Quarkus language server.  
+
+Step 2. Quarkus langauge server checks its cache if completion options
+for the current Quarkus project exists.  
+* If it exists the Quarkus language server sends them to VSCode 
+as the response to the `textDocument/completion` request.
+Communication is complete, and does not proceed to Step 3 
+and onwards.
+* If it does not exist, the Quarkus langauge server sends a
```
Comment (fbricon) [inline CONTRIBUTING.md]: language

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+Step 3. vscode-quarkus receives the `quarkus/projectInfo` request,
+and runs the `quarkus.java.projectInfo` command. This command was
+defined by the Quarkus jdt.ls extension.  
+
+Step 4. The Quarkus jdt.ls extension receives the command, determines
+information about the currently opened Quarkus project and returns
+the information to vscode-quarkus.  
+
+Step 5. vscode-quarkus receives the project information and sends it
+to the Quarkus langauge server.
```
Comment (fbricon) [inline CONTRIBUTING.md]: language

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+defined by the Quarkus jdt.ls extension.  
+
+Step 4. The Quarkus jdt.ls extension receives the command, determines
+information about the currently opened Quarkus project and returns
+the information to vscode-quarkus.  
+
+Step 5. vscode-quarkus receives the project information and sends it
+to the Quarkus langauge server.  
+
+Step 6. Quarkus langauge server receives the information, adds it
```
Comment (fbricon) [inline CONTRIBUTING.md]: language

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -0,0 +1,146 @@
...
+
+### Running vscode-quarkus
+Step 1. Open `vscode-quarkus/` in VSCode.  
+
+Step 2. Open the Debugging tab, select and run 
+"Launch Extension (vscode-quarkus)" at the top left.
+![](images/runExtension.png)
+
+## Debugging  
+### Debugging the Quarkus langauge server:
```
Comment (fbricon) [inline CONTRIBUTING.md]: language

**Review by fbricon — CHANGES_REQUESTED** [review]
That's pretty good! A bunch of typos to fix though

**Comment by xorye** [comment]
Thanks @fbricon, I made the changes.

**Comment by angelozerr** [comment]
I follow the documentation and it works like a charm, well done @xorye !

**Pull request merged**